packages/xstate-wallet/src/constants.ts

```diff
@@ -10,3 +12,19 @@ export const CHALLENGE_DURATION = 0x12c; // 5 minutes
 export const NETWORK_ID = process.env.CHAIN_NETWORK_ID || '0';
 
 export const ETH_TOKEN = '0x0';
+
+export function assetHolderAddress(tokenAddress: string): string | undefined {
+  if (bigNumberify(tokenAddress).isZero()) {
+    return ETH_ASSET_HOLDER_ADDRESS;
+  }
+  // TODO: store mapping, implement lookup
```

Comment:
How is this supposed to work when we are supporting multiple asset holders for the token?

Reply:
The mapping needn't be one-to-one. We could store a mapping from asset holder address to token address, and a mapping from token address to the default asset holder address that our wallet would propose for that token.

This said, I'm not sure (?) if it makes sense to ever support multiple asset holders for a single token within a wallet. The overriding motivation in switching to `assetHolderAddress` is not to support multiple holders per token, but to be able to quickly detect if two wallets are pointing at different asset holders.

Thinking about it now, there isn't actually a danger of two wallets getting to the point where they deposit into different asset holders, as if they use different asset holder addresses, the state hashes won't match, and it will look like the state is not correctly signed. Regardless of this, it seems good to send the asset holder address in the inter-wallet states, as it (a) keeps the property that the state contains all the information required to verify the signature, and (b) it allows the different asset holder situation to be easily detected.